node/heartbeat/monitorEdgeCases_test.go

```diff
@@ -0,0 +1,483 @@
+
+	timer.SetSeconds(hundredSeconds)
+	mon2 := createMonitor(storer, genesisTime, unresponsiveDuration, timer)
+
+	timer.SetSeconds(hundredFiftySeconds)
+	heartBeats = mon2.GetHeartbeats()
+	verifyHeartBeat(t, heartBeats[0], false, 0, 50)
+
+	timer.SetSeconds(twoHundredSeconds)
+	heartBeats = mon2.GetHeartbeats()
+	verifyHeartBeat(t, heartBeats[0], false, 0, 100)
+}
+
+// v: |_________________________________
+// o: |___________|.........|___________
+//    0    10    20        100   150   200
+func TestMonitor_ObserverGapValidatorOnline(t *testing.T) {
+	t.Parallel()
+
+	storer, _ := storage.NewHeartbeatDbStorer(mock.NewStorerMock(), &mock.MarshalizerFake{})
+
+	timer := &mock.MockTimer{}
+	genesisTime := timer.Now()
+
+	mon1 := createMonitor(storer, genesisTime, unresponsiveDuration, timer)
+	mon1.AddHeartbeatMessageToMap(&heartbeat.Heartbeat{Pubkey: []byte(pkValidator)})
+
+	heartBeats := mon1.GetHeartbeats()
+	verifyHeartBeat(t, heartBeats[0], true, 0, 0)
+
+	timer.SetSeconds(tenSeconds)
+	//mon1.AddHeartbeatMessageToMap(&heartbeat.Heartbeat{Pubkey: []byte(pkValidator)})
+	heartBeats = mon1.GetHeartbeats()
+	verifyHeartBeat(t, heartBeats[0], true, 10, 0)
+
+	timer.SetSeconds(twentySeconds)
+	mon1.AddHeartbeatMessageToMap(&heartbeat.Heartbeat{Pubkey: []byte(pkValidator)})
+	heartBeats = mon1.GetHeartbeats()
+	verifyHeartBeat(t, heartBeats[0], true, 20, 0)
+
+	timer.SetSeconds(hundredSeconds)
+	mon2 := createMonitor(storer, genesisTime, unresponsiveDuration, timer)
+	mon2.AddHeartbeatMessageToMap(&heartbeat.Heartbeat{Pubkey: []byte(pkValidator)})
+	heartBeats = mon2.GetHeartbeats()
+	verifyHeartBeat(t, heartBeats[0], true, 100, 0)
+
+	timer.SetSeconds(hundredFiftySeconds)
+	mon2.AddHeartbeatMessageToMap(&heartbeat.Heartbeat{Pubkey: []byte(pkValidator)})
+	heartBeats = mon2.GetHeartbeats()
+	verifyHeartBeat(t, heartBeats[0], true, 150, 0)
+
+	timer.SetSeconds(twoHundredSeconds)
+	mon2.AddHeartbeatMessageToMap(&heartbeat.Heartbeat{Pubkey: []byte(pkValidator)})
+	heartBeats = mon2.GetHeartbeats()
+	verifyHeartBeat(t, heartBeats[0], true, 200, 0)
+}
+
+// v: |_____................
+// o: |_____________________
+//    0    10    60  61   70
+func TestMonitor_ObserverGapValidatorActiveUnitlMaxPeriodEnds(t *testing.T) {
+	t.Parallel()
+
+	storer, _ := storage.NewHeartbeatDbStorer(mock.NewStorerMock(), &mock.MarshalizerFake{})
+
+	timer := &mock.MockTimer{}
+	genesisTime := timer.Now()
+
+	mon1 := createMonitor(storer, genesisTime, unresponsiveDuration, timer)
+	mon1.AddHeartbeatMessageToMap(&heartbeat.Heartbeat{Pubkey: []byte(pkValidator)})
+
+	heartBeats := mon1.GetHeartbeats()
+	verifyHeartBeat(t, heartBeats[0], true, 0, 0)
+
+	timer.SetSeconds(tenSeconds)
+	mon1.AddHeartbeatMessageToMap(&heartbeat.Heartbeat{Pubkey: []byte(pkValidator)})
+	heartBeats = mon1.GetHeartbeats()
+	verifyHeartBeat(t, heartBeats[0], true, 10, 0)
+
+	timer.SetSeconds(60)
+	//mon1.AddHeartbeatMessageToMap(&heartbeat.Heartbeat{Pubkey: []byte(pkValidator)})
```

Comment:
There is commented code in different places in this file. Please remove it.

Reply:
There is still some commented code left on lines 240-242. Please remove.

Reply:
Fixed